pkg/mcp/mcp.go

```diff
@@ -165,3 +167,10 @@ func contextFunc(ctx context.Context, r *http.Request) context.Context {
 
 	return ctx
 }
+
+func toolCallLoggingMiddleware(next server.ToolHandlerFunc) server.ToolHandlerFunc {
+	return func(ctx context.Context, ctr mcp.CallToolRequest) (*mcp.CallToolResult, error) {
```

Comment:
Based on this commit https://github.com/mark3labs/mcp-go/commit/baa715326faba8f95aa371486d72f7665c152892, `CallToolRequest` contains headers. Would you think it would be useful logging some of the values in here like user-agent, etc.?. Would it be useful for us or unnecessary noise?

Reply:
Actually this middleware https://github.com/manusa/kubernetes-mcp-server/blob/775fa21bd1345eee52fea6089c119821c0d95c9e/pkg/http/middleware.go#L12 logs all the requests (if verbosity is set to 5 or higher). Maybe we can log the headers in there (only the ones that are not carrying sensitive data).

For what this middleware will be useful for debugging cases?. Request is seen in the audit logs and verifying that it actually results in tool call?

Reply:
> Actually this middleware...
This works for HTTP but not for STDIO (for example).

IMO is good to have both and also keep them separate.
An HTTP call might eventually become a tool call or not, especially now that we're working on auth (e.g. we get an HTTP request but auth blocks it).

> In which scenario this middleware will be useful?. 

The user who made the request was building an MCP proxy and wasn't able to see any effects on the MCP server. They wanted to know if the tool was actually being called.
There are multiple places where the calls can be logged (MCP Client, MCP Host, etc.), I guess that this is yet another tracing point.
I understand it can be interesting to trace where the tool call chain was broken.